inc/classes/class-service-worker.php

```diff
@@ -97,6 +98,97 @@ public function cache_images( \WP_Service_Worker_Scripts $scripts ) {
 
 	}
 
+	/**
+	 * Pre-Cache latest blog posts
+	 *
+	 * @param \WP_Service_Worker_Scripts $scripts Instance to register service worker behavior with.
+	 *
+	 * @return void
+	 */
+	public function precache_latest_blog_posts( \WP_Service_Worker_Scripts $scripts ) {
+
+		$recent_posts = new \WP_Query(
+			array(
+				'post_type'              => 'post',
+				'post_status'            => 'publish',
+				'fields'                 => 'ids',
+				'posts_per_page'         => 10,
+				'no_found_rows'          => true,
+				'update_post_meta_cache' => false,
+				'update_post_term_cache' => false,
+			)
+		);
+
+		if ( empty( $recent_posts->posts ) || ! is_array( $recent_posts->posts ) ) {
+			return;
+		}
+
+		foreach ( $recent_posts->posts as $recent_post_id ) {
+			$scripts->precaching_routes()->register(
+				get_permalink( $recent_post_id ),
+				array(
+					'revision' => get_bloginfo( 'version' ),
+				)
+			);
+		}
+
+	}
+
+	/**
+	 * Pre-Cache menu
+	 *
+	 * Only precache menu which assigned to any menu locations.
+	 *
+	 * @param \WP_Service_Worker_Scripts $scripts Instance to register service worker behavior with.
+	 *
+	 * @return void
+	 */
+	public function precache_menu( \WP_Service_Worker_Scripts $scripts ) {
+
+		// Get menu locations from source site.
+		$menu_locations = get_nav_menu_locations();
```

Comment:
Same here. Caching will be required to not let it run for every page load. @chandrapatel